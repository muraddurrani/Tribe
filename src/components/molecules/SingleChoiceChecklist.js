import React, { useEffect, useState } from 'react'
import { FlatList, StyleSheet, View } from 'react-native'
import { CheckBox } from 'react-native-elements'

import colours from '../../styles/colours'

function SingleChoiceChecklist({data, title, width, height, onCheck, initChecked, style, ...props}) {
  const [checked, setChecked] = useState()

  const handleCheck = (item) => {
    setChecked(item.id)
  }

  const render = ({item}) => (
    <CheckBox
      containerStyle = {styles.checkbox}
      title = {title ? title : item.name}
      textStyle = {styles.title}
      checked = {checked == item.id}
      checkedColor = {colours.midpoint3}
      onPress = {() => {
        handleCheck(item)
        onCheck(item)
      }}
    />
  )

  useEffect(() => {
    if (initChecked) {
      setChecked(initChecked)
    }
  }, [initChecked])

  return (
    <View style = {{height, width, ...style}}>
      <FlatList
        {...props}
        showsVerticalScrollIndicator = {false}
        data = {data}
        keyExtractor = {(item, index) => index}
        renderItem = {render}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  checkbox: {
    backgroundColor: 'transparent',
    borderColor: 'transparent'
  },
  title: {
    color: colours.gray6,
    fontWeight: '400',
    textTransform: 'capitalize'
  }
})

export default SingleChoiceChecklist
