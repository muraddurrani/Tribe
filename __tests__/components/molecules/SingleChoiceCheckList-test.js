import React from 'react';
import renderer from 'react-test-renderer';
import SingleChoiceChecklist from '../../../src/components/molecules/SingleChoiceChecklist'

jest.useFakeTimers()

it('SingleChoiceCheckList renders correctly', () => {
  const tree = renderer
    .create(
    <SingleChoiceChecklist 
      title = 'Gender'
      data = {['Male', 'Female', 'Others', 'Preferred not say']}
      onCheck = {console.log('single checked')}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});