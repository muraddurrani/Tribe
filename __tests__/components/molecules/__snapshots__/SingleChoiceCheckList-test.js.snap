// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleChoiceCheckList renders correctly 1`] = `
<View
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        "Male",
        "Female",
        "Others",
        "Preferred not say",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            Object {
              "checked": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="checkbox"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FF7D8E",
                    "fontSize": 24,
                  },
                  Object {
                    "minWidth": 24,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#6e6e6e",
                  "fontWeight": "400",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textTransform": "capitalize",
                }
              }
              testID="checkboxTitle"
            >
              Gender
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            Object {
              "checked": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="checkbox"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FF7D8E",
                    "fontSize": 24,
                  },
                  Object {
                    "minWidth": 24,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#6e6e6e",
                  "fontWeight": "400",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textTransform": "capitalize",
                }
              }
              testID="checkboxTitle"
            >
              Gender
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            Object {
              "checked": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="checkbox"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FF7D8E",
                    "fontSize": 24,
                  },
                  Object {
                    "minWidth": 24,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#6e6e6e",
                  "fontWeight": "400",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textTransform": "capitalize",
                }
              }
              testID="checkboxTitle"
            >
              Gender
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            Object {
              "checked": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="checkbox"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FF7D8E",
                    "fontSize": 24,
                  },
                  Object {
                    "minWidth": 24,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#6e6e6e",
                  "fontWeight": "400",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textTransform": "capitalize",
                }
              }
              testID="checkboxTitle"
            >
              Gender
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
